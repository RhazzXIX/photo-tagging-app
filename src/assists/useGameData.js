import { useState, useEffect } from "react";
import { database, storage } from "./fbConfig";
import { getDownloadURL, ref } from "firebase/storage";
import { getDoc, doc } from "firebase/firestore";

export default function useGameData(props) {
  const [firstMap, setFirstmap] = useState({});
  const [scndMap, setScndMap] = useState({});
  const [charSelection1, setCharSelection1] = useState([]);
  const [charSelection2, setCharSelection2] = useState([]);

  const selection1Ref = doc(database, "game", "anime");
  const selection2Ref = doc(database, "game", "game");

  const getData = async () => {
    const firstMapData = {};
    const scndMapData = {};
    const char1 = {};
    const char2 = {};
    const char3 = {};
    const char4 = {};
    const char5 = {};
    const char6 = {};

    const data1 = await getDoc(selection1Ref).then((response) =>
      response.data()
    );
    firstMapData.name = data1.map.name;
    await getDownloadURL(ref(storage, data1.map.imgRef)).then(
      (url) => (firstMapData.url = url)
    );
    setFirstmap(firstMapData);
    char1.name = data1.char1.name;
    await getDownloadURL(ref(storage, data1.char1.imgRef)).then(
      (url) => (char1.url = url)
    );
    char2.name = data1.char2.name;
    await getDownloadURL(ref(storage, data1.char2.imgRef)).then(
      (url) => (char2.url = url)
    );
    char3.name = data1.char3.name;
    await getDownloadURL(ref(storage, data1.char3.imgRef)).then(
      (url) => (char3.url = url)
    );
    setCharSelection1([char1, char2, char3]);

    const data2 = await getDoc(selection2Ref).then((response) =>
      response.data()
    );
    scndMapData.name = data2.map.name;
    await getDownloadURL(ref(storage, data2.map.imgRef)).then(
      (url) => (scndMapData.url = url)
    );
    setScndMap(scndMapData);
    char4.name = data2.char1.name;
    await getDownloadURL(ref(storage, data2.char1.imgRef)).then(
      (url) => (char4.url = url)
    );
    char5.name = data2.char2.name;
    await getDownloadURL(ref(storage, data2.char2.imgRef)).then(
      (url) => (char5.url = url)
    );
    char6.name = data2.char3.name;
    await getDownloadURL(ref(storage, data2.char3.imgRef)).then(
      (url) => (char6.url = url)
    );
    setCharSelection2([char4, char5, char6]);
  };

  useEffect(() => {
    getData();
  }, []);

  return { firstMap, scndMap, charSelection1, charSelection2 };
}
