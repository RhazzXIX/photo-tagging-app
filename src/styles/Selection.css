main#selection {
  flex: auto;
  top: 0;
  background-color: #22107f;
  display: flex;
  justify-content: space-evenly;
}

div.card {
  margin-top: 180px;
  width: 400px;
  height: 300px;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  background-color: rgba(255, 255, 255, 0.9);
  border-radius: 12px;
  box-shadow: 0 0 10px black;
}

div.card:hover {
  box-shadow: 0 0 10px white;
}

div.card ul {
  display: flex;
  justify-content: space-evenly;
  width: 100%;
}

div.card ul li {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 4px;
}

div.card > img {
  width: 180px;
  box-shadow: 0 0 4px black;
  border-radius: 8px;
}

div.card > button {
  padding: 4px 8px;
  border-radius: 10px;
  font-size: 1.2rem;
}
