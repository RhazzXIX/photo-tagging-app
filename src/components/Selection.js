import { Link } from "react-router-dom";
import useGameData from "../assists/useGameData";
import "../styles/Selection.css";

export default function Selection(props) {
  const { firstMap, scndMap, charSelection1, charSelection2 } = useGameData();

  return (
    <main id="selection">
      <div className="card">
        <ul>
          {charSelection1.map((char) => {
            return (
              <li key={char.name}>
                <img src={char.url} alt={char.name} className="chars" />
                <p>{char.name}</p>
              </li>
            );
          })}
        </ul>
        <img src={firstMap.miniUrl} alt={firstMap.name} />
        <Link to="animeX">
          <button>{firstMap.name}</button>
        </Link>
      </div>
      <div className="card">
        <ul>
          {charSelection2.map((char) => {
            return (
              <li key={char.name}>
                <img src={char.url} alt={char.name} className="chars" />
                <p>{char.name}</p>
              </li>
            );
          })}
        </ul>
        <img src={scndMap.miniUrl} alt={scndMap.name} />
        <Link to="gameX">
          <button>{scndMap.name}</button>
        </Link>
      </div>
    </main>
  );
}
