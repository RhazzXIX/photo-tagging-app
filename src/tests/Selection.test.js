import "@testing-library/jest-dom";
import useGameData from "../assists/useGameData";
import { render } from "@testing-library/react";
import Selection from "../components/Selection";
import { BrowserRouter } from "react-router-dom";

const fetchedData = {
  firstMap: {
    miniUrl: "mini url 1",
    name: "Anime Crossover",
    url: "url 2",
  },
  scndMap: {
    miniUrl: "mini url 2",
    name: "Game Crossover",
    url: "url 2",
  },
  charSelection1: [
    {
      name: 'Yato "Yaboku"',
      url: "yato url",
    },
    {
      name: 'ELUCIA"',
      url: "elucia url",
    },
    {
      name: 'EVANS"',
      url: "evans url",
    },
  ],
  charSelection2: [
    {
      name: "Spyro",
      url: "spyro url",
    },
    {
      name: 'Arthur"',
      url: "arthur url",
    },
    {
      name: 'Marco"',
      url: "marco url",
    },
  ],
};

jest.mock("../assists/useGameData", () => {
  return () => fetchedData;
});

describe("Selection component", () => {
  it("Renders on screen", () => {
    const { container } = render(
      <BrowserRouter>
        <Selection />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
