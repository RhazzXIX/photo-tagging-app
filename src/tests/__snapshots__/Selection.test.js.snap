// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selection component Renders on screen 1`] = `
<div>
  <main
    id="selection"
  >
    <div
      class="card"
    >
      <ul>
        <li>
          <img
            alt="Yato \\"Yaboku\\""
            class="chars"
            src="yato url"
          />
          <p>
            Yato "Yaboku"
          </p>
        </li>
        <li>
          <img
            alt="ELUCIA\\""
            class="chars"
            src="elucia url"
          />
          <p>
            ELUCIA"
          </p>
        </li>
        <li>
          <img
            alt="EVANS\\""
            class="chars"
            src="evans url"
          />
          <p>
            EVANS"
          </p>
        </li>
      </ul>
      <img
        alt="Anime Crossover"
        src="mini url 1"
      />
      <a
        href="/anime"
      >
        <button>
          Anime Crossover
        </button>
      </a>
    </div>
    <div
      class="card"
    >
      <ul>
        <li>
          <img
            alt="Spyro"
            class="chars"
            src="spyro url"
          />
          <p>
            Spyro
          </p>
        </li>
        <li>
          <img
            alt="Arthur\\""
            class="chars"
            src="arthur url"
          />
          <p>
            Arthur"
          </p>
        </li>
        <li>
          <img
            alt="Marco\\""
            class="chars"
            src="marco url"
          />
          <p>
            Marco"
          </p>
        </li>
      </ul>
      <img
        alt="Game Crossover"
        src="mini url 2"
      />
      <a
        href="/game"
      >
        <button>
          Game Crossover
        </button>
      </a>
    </div>
  </main>
</div>
`;
